Bugfixes and implemented an error catch block to utils func

Keilvin opened the pull request.

Keilvin committed:
Minor bugfix and Implemented error catching for getDiffTimeFormatted in utils

Bugfix csv = csv =  csv on line 403
in sessions.py

Implemented error catching to not allow negative time difference in getTimeDiffFormatted, should hopefully catch NaN

Keilvin committed:
Merge branch 'main' of https://github.com/Keilvin/ELLE-2023-Website-API

Keilvin committed:
Enabled Authentication on all Mentor endpoints

Keilvin committed:
Added more permissions

Allowing SU to add classes

Keilvin committed:
Bugfixes for some get requests

Namely GetMentorQuestions and GetMultipleChoiceOptions

Keilvin committed:
Update sessions.py

Outer indentation error

JustinRMoy commented on resources/mentors.py:
```diff
@@ -184,15 +185,16 @@ def post(self):
+    @jwt_required
 
     # @jwt_required
-    def post(self):
```
this needs to be a post and NOT a get. This will ruin Unity's ability to interact with the server

JustinRMoy commented on resources/mentors.py:
```diff
@@ -319,14 +321,14 @@ def delete(self):
 
 class GetMultipleChoiceOptions(Resource):
-    # @jwt_required
-    def post(self):
```
same here, these were split off and made posts for the reason stated above

Keilvin replied:
Got it

JustinRMoy commented on utils.py:
```diff
@@ -389,15 +389,24 @@ def getTimeDiffFormatted(time_1 = None,
         d = {"days": time_delta.days}
-        d["hours"], rem = divmod(time_delta.seconds, 3600)
+        hours, rem = divmod(time_delta.seconds, 3600)
+        d["hours"] = hours
```
what's the point of this change?

Keilvin replied:
I think here it's just easier to understand

JustinRMoy replied:
It's  a pointless change and unnecessarily creates a new variable while not really making a real change as the values are still assigned the same way but not with extra steps

Keilvin replied:
ok

JustinRMoy reviewed (commented):
Undo the changes of making Post requests Get. Adress the other comments

JustinRMoy commented on resources/group.py:
```diff
@@ -26,7 +26,7 @@ def post(self):
             cursor = conn.cursor()
 
-            if permission != 'pf':
+            if permission != 'pf' or permission != 'su':
```
Why would students be able to create classes? This sounds like something only a professor should be able to do

Keilvin replied:
ST stands for Student, PF stands for Professor and SU stands for Super User

JustinRMoy replied:
I see

JustinRMoy reviewed (commented):
This commit specifically should be undone entirely. These should remain as Posts, you should have seen when I made this change from Get to Post when you did my code review

```
[Bugfixes for some get requests] 
[5aaf4df]
Namely GetMentorQuestions and GetMultipleChoiceOptions
```

Keilvin committed:
Update mentors.py

Switched the getters back to post for Unity implementation

JustinRMoy reviewed (changes requested):
address my last comment and it should be good to go

Keilvin committed:
Update utils.py

Addressed previous comment, reverted line 408/409

Keilvin committed:
Sessions and Utils

Fixed download of CSV with changes in code, also unsure why mullti day is negative, the database cannot track multi-day anyway because it only tracks hours in 24 hour time.

JustinRMoy commented on utils.py:
```diff
@@ -403,11 +403,14 @@ def getTimeDiffFormatted(time_1 = None,
         d = {"days": time_delta.days}
         d["hours"], rem = divmod(time_delta.seconds, 3600)
         d["minutes"], d["seconds"] = divmod(rem, 60)
-        if d['days'] != 1 and d['days'] != -1:
```
why the change from AND to OR?

Keilvin committed:
Update utils.py

Commented out the line for now since we aren't using days until I can figure out why it is negative days
